core/src/main/scala/akka/kafka/scaladsl/MetadataClient.scala

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright (C) 2014 - 2016 Softwaremill <http://softwaremill.com>
+ * Copyright (C) 2016 - 2019 Lightbend Inc. <http://www.lightbend.com>
+ */
+
+package akka.kafka.scaladsl
+import akka.actor.{ActorRef, ActorSystem}
+import akka.kafka.{ConsumerSettings, KafkaConsumerActor}
+import akka.kafka.Metadata.{BeginningOffsets, GetBeginningOffsets}
+import akka.pattern.ask
+import akka.util.Timeout
+import org.apache.kafka.common.TopicPartition
+
+import scala.concurrent.{ExecutionContext, Future}
+
+object MetadataClient {
+
+  private val consumerActors = scala.collection.mutable.Map[ConsumerSettings[Any, Any], ActorRef]()
```

Comment:
It should be left to the API user to create and track the consumer actors. The Kafka client runs a number of threads that conscious users might want to avoid.

Reply:
I removed it. Please look at it again and tell me if this is a right way.